File: JAVA/Menu.java
import java.util.Scanner;

public class Menu {
    public static void displayMenu() {
        System.out.println("\nMenu:");
        System.out.println("1. Add Money");
        System.out.println("2. Withdraw Money");
        System.out.println("3. Show Balance");
        System.out.println("4. Display History");
        System.out.println("5. Display Cards Info");
        System.out.println("6. Withdraw ID Card");
        System.out.println("7. Withdraw Driving License");
        System.out.println("8. Put Back ID Card");
        System.out.println("9. Put Back Driving License");
        System.out.println("10. Exit");
    }

    public static void startMenu(Wallet wallet) {
        Scanner scanner = new Scanner(System.in);

        while (true) {
            displayMenu();
            System.out.print("Select an option (1-10): ");
            String option = scanner.nextLine();

            switch (option) {
                case "1":
                    System.out.print("Enter the amount to add: ");
                    double amountToAdd = Double.parseDouble(scanner.nextLine());
                    wallet.addMoney(amountToAdd);
                    break;
                case "2":
                    System.out.print("Enter the amount to withdraw: ");
                    double amountToWithdraw = Double.parseDouble(scanner.nextLine());
                    wallet.withdrawMoney(amountToWithdraw);
                    break;
                case "3":
                    wallet.showBalance();
                    break;
                case "4":
                    wallet.displayHistory();
                    break;
                case "5":
                    wallet.displayCardsInfo();
                    break;
                case "6":
                    wallet.withdrawCard("ID Card");
                    break;
                case "7":
                    wallet.withdrawCard("Driving License");
                    break;
                case "8":
                    wallet.putBackCard("ID Card");
                    break;
                case "9":
                    wallet.putBackCard("Driving License");
                    break;
                case "10":
                    scanner.close();
                    break;
                default:
                    System.out.println("Invalid option. Please enter a number from 1 to 10.");
            }
        }
    }
}

File: JAVASCRIPT/menu.js
function displayMenu() {
  console.log("\nMenu:");
  console.log("1. Add Money");
  console.log("2. Withdraw Money");
  console.log("3. Show Balance");
  console.log("4. Display History");
  console.log("5. Display Cards Info");
  console.log("6. Withdraw ID Card");
  console.log("7. Withdraw Driving License");
  console.log("8. Put Back ID Card");
  console.log("9. Put Back Driving License");
  console.log("10. Exit");
}

function startMenu(wallet, rl) {
  displayMenu();
  rl.question("Select an option (1-10): ", function (option) {
    switch (option) {
      case '1':
        rl.question("Enter the amount to add: ", function (amount) {
          wallet.addMoney(parseFloat(amount));
          startMenu(wallet, rl);
        });
        break;

      case '2':
        rl.question("Enter the amount to withdraw: ", function (amount) {
          wallet.withdrawMoney(parseFloat(amount));
          startMenu(wallet, rl);
        });
        break;

      case '3':
        wallet.showBalance();
        startMenu(wallet, rl);
        break;

      case '4':
        wallet.displayHistory();
        startMenu(wallet, rl);
        break;

      case '5':
        wallet.displayCardsInfo();
        startMenu(wallet, rl);
        break;

      case '6':
        wallet.withdrawCard("ID Card");
        startMenu(wallet, rl);
        break;

      case '7':
        wallet.withdrawCard("Driving License");
        startMenu(wallet, rl);
        break;

      case '8':
        wallet.putBackCard("ID Card");
        startMenu(wallet, rl);
        break;

      case '9':
        wallet.putBackCard("Driving License");
        startMenu(wallet, rl);
        break;

      case '10':
        rl.close();
        break;

      default:
        console.log("Invalid option. Please enter a number from 1 to 10.");
        startMenu(wallet, rl);
        break;
    }
  });
}

module.exports = { displayMenu, startMenu };
